1주차

wlaud2000 opened the pull request:
**1주차 과제입니다!**
각 요청에 맞게 Http Method이용하여 Controller작성했습니다.

감사합니다✨


사용한 Annotation과 알게 된 것
---
### 1. `@RequestBody`

`@RequestBody` 는 HttpMessageConverter가 HTTP Request Body내의 데이터를 객체로 변환하도록 시키는 Annotation이다.

HTTP 요청의 바디내용을 통째로 자바객채로 변환해서 매핑된 메소드 파라미터로 전달한다!

### 2. 요청방식들의 위치
`@RequestMapping`은 Class와 Method에 붙일 수 있고, `@GetMapping, @PostMapping, @PutMapping, @DeleteMapping` 등은 Method에만 붙일 수 있다!

### 3. `@Slf4j`
Simple Logging Facade for Java의 약자로 다양한 로깅 프레임워크에 대한 인터페이스 역할을 하는 라이브러리이다!
* _Facade는 보통 건축 분야에서 건물의 정면 외벽을 지칭하는 말인데, 소프트웨어 분야에서는 [Facade Pattern]이라는 디자인 패턴에서 사용되는 인터페이스를 지칭할 때 쓰인다._

**로그레벨**
심각도 수준 = Error > Warn > Info > Debug > Trace

|레벨|설명|
|---|-----|
|Error|요청을 처리하는 중 문제가 발생한 상태를 나타냄|
|Warn|처리 가능한 문제이지만 향후 에러가 될 수 있는 경고성 메세지를 나타냄|
|Info|명확한 의도가 있는 정보성(로그인, 상태변경 등등...) 메세지를 나타냄|
|Debug|개발 시 디버그 용도로 사용한 메세지를 나타냄
|Trace|Debug레벨보다 더 상세한 상태를 나타냄 |

DDonghyeo committed:
1주차 과제

wlaud2000 committed:
1주차 과제

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/OrderController.java:
```diff
@@ -0,0 +1,47 @@
+    @PostMapping(value = "")
+    public String createOrder(@RequestBody CreateOrderRequestDto createOrderRequestDto){
+        log.info("주문 생성하기");
+        log.info("name = {}", createOrderRequestDto.getName());
```
Slfj4 제대로 사용하네요 !! 굿👍

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/OrderController.java:
```diff
@@ -0,0 +1,47 @@
+    // 3. 주문을 수정하는 컨트롤러를 만듭니다. 이때 return 값은 "주문 수정하기"입니다.
+    //UpdateOrderRequestDto 클래스를 매개변수로 받습니다.
+    @PutMapping(value = "")
+    public String updateOrder(@RequestBody UpdateOrderRequestDto updateOrderRequestDto) {
```
현재 UpdateOrderRequestDto에 Order의 id값이 없습니다 ~~ dto에 id를 추가하거나 다른 방법으로 id의 파라미터를 추가로 받아주세요~!!

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/UserController.java:
```diff
@@ -0,0 +1,51 @@
+    // 이때 log.info를 이용를여 사용자의 이름, 주소를 출력해줍니다. return 값은 "사용자 수정"입니다.
+    //UpdateUserRequestDto 클래스를 매개변수로 받습니다.
+    @PutMapping(value = "")
+    public String updateUser(@RequestBody UpdateUserRequestDto updateUserRequestDto) {
```
UpdateUserRequestDto도 마찬가지로 id가 없습니다 !! 어떤 id의 user를 업데이트할지 알려줘야합니다 !! 마찬가지로 dto에 필드를 추가하거나 파라미터 추가로 받아주세요 ~~

-------------------
보통 RUD에는 조회, 업데이트, 삭제의 "대상"이 필요한데, DB내에서 Unique한 값으로 식별자를 파라미터로 전달합니다. 보통은 DB의 Primary Key인 id를 사용하거나, Unique한 값인 User의 Email을 사용하게 됩니다.

C (Create)의 경우에는 id를 자동으로 생성하기 때문에 필요하지 않습니다.

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/UserController.java:
```diff
@@ -0,0 +1,51 @@
+    // 4. 사용자를 삭제하는 컨트롤러를 만듭니다.
+    // 이때 log.info 이용하여 사용자의 ID를 출력해줍니다. return 값은 "사용자 삭제"입니다.
+    @DeleteMapping(value = "")
+    public String deleteUser(@RequestBody CreateUserRequestDto createUserRequestDto) {
```
삭제할 User의 대상은 User 의 Id (식별자)만 있으면 됩니다. 파라미터를 userId로 받아주세요~~

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/UserController.java:
```diff
@@ -0,0 +1,51 @@
+    // 2. 사용자를 조회하는 컨트롤러를 만듭니다.
+    // 이때 log.info를 이용하여 사용자의 ID를 출력해줍니다. return 값은 "사용자 조회"입니다.
+    @GetMapping(value = "")
+    public String getUser(@RequestBody CreateUserRequestDto createUserRequestDto) {
```
조회할 User의 대상은 UserId만 있으면 됩니다. 파라미터를 userId로 받아주세요 ~~

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/OrderController.java:
```diff
@@ -0,0 +1,47 @@
+
+    // 2. 주문을 가져오는 컨트롤러를 만듭니다. 이때 return 값은 "주문 가져오기"입니다.
+    @GetMapping(value = "")
+    public String getOrder() {
```
주문을 가져올 대상을 파라미터로 받아야합니다. orderId로 받아주세요~~

DDonghyeo commented on src/main/java/com/likelion/lionshop/controller/OrderController.java:
```diff
@@ -0,0 +1,47 @@
+
+    // 4. 주문을 삭제하는 컨트롤러를 만듭니다. 이때 return 값은 "주문 삭제하기"입니다.
+    @DeleteMapping(value = "")
+    public String deleteOrder() {
```
마찬가지로 OrderId 파라미터로 받아주세요~~

DDonghyeo reviewed (approved):
수고하셨습니다 👍

DDonghyeo reviewed (changes requested):
파라미터 수정해주세요 ~~

wlaud2000 committed:
"fix: 파라미터 수정 및 불필요한 코드 삭제"

*파라미터에서 불필요한 @RequestBody 코드를 삭제하였습니다.
*CreateOrderRequestDto와 UpdateOrderRequestDto에 Long id 추가하였습니다.

wlaud2000 commented:
파라미터 수정 완료했습니다!

DDonghyeo reviewed (approved):
수고하셨습니다 👍
